File: lib/screens/terms_screen.dart
import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';
import '../theme.dart';

class TermsAndConditionsScreen extends StatelessWidget {
  const TermsAndConditionsScreen({super.key});

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Theme.of(context).scaffoldBackgroundColor,
      appBar: AppBar(
        title: Semantics(
          header: true,
          child: const Text('Terms and Conditions'),
        ),
        centerTitle: true,
        elevation: 0,
        backgroundColor: Colors.transparent,
      ),
      body: SafeArea(
        child: SingleChildScrollView(
          padding: const EdgeInsets.all(20),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            children: const [
              _Heading('Terms and Conditions'),
              _Body(
                'Geeta Nexus is provided by Satvik Technologys, an independent developer entity led by Kuldeep Kumar Yadav. By installing and using the app, you agree to the following terms.',
              ),
              _Heading('1. Use of the App'),
              _Body(
                'Geeta Nexus is provided for personal, non-commercial spiritual and educational use. You agree to use the app respectfully and lawfully.',
              ),
              _Heading('2. Spiritual Content'),
              _Body(
                'Translations, summaries, and AI-generated answers are presented for inspiration and study. They are not a substitute for guidance from a qualified spiritual teacher.',
              ),
              _Heading('3. AI-Generated Responses'),
              _Body(
                'AI features may occasionally produce inaccurate or incomplete information. Always use your own judgement.',
              ),
              _Heading('4. Intellectual Property'),
              _Body(
                'The Bhagavad Gita is in the public domain. App design, code, UI, and original content are the property of Satvik Technologys.',
              ),
              _Heading('5. Limitation of Liability'),
              _Body(
                'The app is provided "as is" without warranties of any kind. Satvik Technologys and the developer are not liable for any decisions made based on the content of the app.',
              ),
              _Heading('6. Changes to Terms'),
              _Body(
                'These terms may be updated from time to time. Continued use of the app means you accept the latest version.',
              ),
              _Heading('7. Governing Context'),
              _Body(
                'These terms are intended to be interpreted in line with applicable laws of India.',
              ),
              _Heading('8. Contact'),
              _Body(
                'For questions about these terms, contact Satvik Technologys:\nEmail: kuldeepky538@gmail.com\nAddress: Korba, Chhattisgarh, India.',
              ),
              SizedBox(height: 30),
            ],
          ),
        ),
      ),
    );
  }
}

class _Heading extends StatelessWidget {
  final String text;
  const _Heading(this.text);

  @override
  Widget build(BuildContext context) => Padding(
        padding: const EdgeInsets.only(top: 18, bottom: 8),
        child: Semantics(
          header: true,
          child: Text(
            text,
            style: GoogleFonts.cinzel(
              color: Theme.of(context).colorScheme.primary,
              fontSize: 14,
              fontWeight: FontWeight.bold,
              letterSpacing: 1.1,
            ),
          ),
        ),
      );
}

class _Body extends StatelessWidget {
  final String text;
  const _Body(this.text);

  @override
  Widget build(BuildContext context) => Padding(
        padding: const EdgeInsets.only(bottom: 8),
        child: Text(
          text,
          style: GoogleFonts.crimsonText(
            color: Theme.of(context).colorScheme.onSurface,
            fontSize: 16,
            height: 1.6,
          ),
        ),
      );
}

File: androidApp/src/main/kotlin/com/nexuswavetech/geetanexus/ui/screens/TermsScreen.kt
package com.nexuswavetech.geetanexus.ui.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.LazyListScope
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.navigation.NavController
import com.nexuswavetech.geetanexus.AppConfig

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun TermsScreen(navController: NavController) {
    Scaffold(
        topBar = {
            TopAppBar(
                title = { Text("Terms of Service", fontWeight = FontWeight.Bold) },
                navigationIcon = {
                    IconButton(onClick = { navController.popBackStack() }) {
                        Icon(Icons.Default.ArrowBack, contentDescription = "Back")
                    }
                }
            )
        }
    ) { padding ->
        LazyColumn(
            modifier       = Modifier.fillMaxSize().padding(padding),
            contentPadding = PaddingValues(16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            item {
                Text("Effective date: June 2025", style = MaterialTheme.typography.labelMedium,
                    color = MaterialTheme.colorScheme.onSurfaceVariant)
            }

            termsSection("1. Acceptance of Terms",
                "By downloading, installing, or using ${AppConfig.APP_NAME} ("the App"), you agree to be bound by these Terms of Service. If you do not agree to these terms, do not use the App.\n\nThe App is operated by ${AppConfig.COMPANY_NAME} ("Company", "we", "us").")

            termsSection("2. Description of Service",
                "${AppConfig.APP_NAME} provides:\n• Digital access to the Bhagavad Gita, Shiva Mahapurana, and Ramcharitmanas\n• An AI-powered spiritual guidance assistant (Aira)\n• Text-to-speech audio narration of sacred texts\n• Personal bookmarking and journaling features\n• Community and devotional resources\n\nThe App is provided free of charge. Premium features may be introduced in future updates.")

            termsSection("3. User Conduct",
                "You agree NOT to:\n• Use the App for any unlawful purpose\n• Attempt to reverse-engineer, decompile, or extract source code\n• Use the AI assistant to generate harmful, offensive, or misleading content\n• Circumvent the security measures (Ed25519 signature verification) protecting the API\n• Impersonate another person or entity\n• Use the App to spam or harass others")

            termsSection("4. Intellectual Property",
                "The sacred texts included (Bhagavad Gita, Shiva Mahapurana, Ramcharitmanas) are in the public domain. The translations and commentary used may be sourced from various open-license providers.\n\nThe App's source code, design, branding, logo, and AI system (Aira) are the intellectual property of ${AppConfig.COMPANY_NAME}. You may not reproduce, distribute, or create derivative works without written permission.\n\nUser-generated content (journal entries, bookmarks) remains your property and is stored only on your device.")

            termsSection("5. AI Disclaimer",
                "Aira, our AI spiritual guide, is powered by Google Gemini and other AI models. While we strive for accuracy:\n\n• AI responses are for informational and inspirational purposes only\n• AI may make errors in interpreting scripture — always consult a qualified guru or scholar for important spiritual decisions\n• The AI does not replace professional advice of any kind (medical, legal, financial, or otherwise)\n• We are not responsible for decisions made based solely on AI-generated responses")

            termsSection("6. Audio Content",
                "Text-to-speech audio is generated on-demand using AI models. The audio:\n• Is for personal, non-commercial use only\n• May not be downloaded, distributed, or used commercially\n• Quality may vary based on AI model performance\n• Requires internet connectivity to generate")

            termsSection("7. Availability & Changes",
                "We reserve the right to:\n• Modify, suspend, or discontinue any part of the App without notice\n• Update these Terms at any time (continued use = acceptance)\n• Change or remove features, including AI capabilities, based on third-party API availability\n\nWe will make reasonable efforts to notify users of significant changes.")

            termsSection("8. Limitation of Liability",
                "${AppConfig.COMPANY_NAME} and its officers, directors, employees, and agents shall not be liable for:\n• Any indirect, incidental, special, or consequential damages\n• Loss of data or profits\n• Errors in AI-generated spiritual guidance\n• Service interruptions\n\nThe App is provided 'as is' without warranties of any kind.")

            termsSection("9. Governing Law",
                "These Terms shall be governed by and construed in accordance with the laws of India. Any disputes shall be subject to the exclusive jurisdiction of courts in India.")

            termsSection("10. Contact",
                "For questions about these Terms:\n\n${AppConfig.COMPANY_NAME}\nEmail: ${AppConfig.COMPANY_EMAIL}\nWebsite: ${AppConfig.Social.WEBSITE}")
        }
    }
}

private fun LazyListScope.termsSection(title: String, body: String) {
    item {
        ElevatedCard(shape = RoundedCornerShape(16.dp)) {
            Column(modifier = Modifier.padding(16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text(title, style = MaterialTheme.typography.titleSmall, fontWeight = FontWeight.Bold)
                Text(body, style = MaterialTheme.typography.bodySmall,
                    color = MaterialTheme.colorScheme.onSurfaceVariant, lineHeight = 22.sp)
            }
        }
    }
}
